tests/integration/integrations/conftest.py

```diff
@@ -5,61 +5,206 @@
 
 import collections
 import logging
-import os
 import pathlib
 import time
 
+import jubilant
 import kubernetes
-import nest_asyncio
 import pytest
-import pytest_asyncio
-from juju.application import Application
 from minio import Minio
-from ops import JujuVersion
-from pytest_operator.plugin import OpsTest
+
+from tests.integration.conftest import build_charm_file
+from tests.integration.types import App
 
 PROJECT_ROOT = pathlib.Path(__file__).parent.parent.parent.parent
-nest_asyncio.apply()
 
 logger = logging.getLogger(__name__)
 
 
-@pytest.fixture(autouse=True)
-def skip_by_juju_version(request, model):
-    """Skip the test if juju version is lower then the `skip_juju_version` marker value."""
-    if request.node.get_closest_marker("skip_juju_version"):
-        current_version = JujuVersion(
-            f"{model.info.agent_version.major}.{model.info.agent_version.minor}.{model.info.agent_version.patch}"
+def deploy_postgresql(
+    juju: jubilant.Juju,
+):
+    """Deploy and set up postgresql charm needed for the 12-factor charm."""
+
+    if juju.status().apps.get("postgresql-k8s"):
+        logger.info("postgresql-k8s already deployed")
+        return
+
+    juju.deploy(
+        "postgresql-k8s",
+        channel="14/stable",
+        base="ubuntu@22.04",
+        revision=300,
+        trust=True,
+        config={"profile": "testing"},
+    )
+    juju.wait(
+        lambda status: status.apps["postgresql-k8s"].is_active,
+        timeout=20 * 60,
+    )
+    juju.config(
+        "postgresql-k8s",
+        {
+            "plugin_hstore_enable": "true",
+            "plugin_pg_trgm_enable": "true",
+        },
+    )
+    juju.wait(lambda status: status.apps["postgresql-k8s"].is_active)
+
+
+@pytest.fixture(scope="function", name="flask_app")
+def flask_app_fixture(juju: jubilant.Juju, pytestconfig: pytest.Config):
+    framework = "flask"
+    yield next(
+        generate_app_fixture(
+            juju=juju,
+            pytestconfig=pytestconfig,
+            framework=framework,
+            image_name=f"test-{framework}-image",
+            use_postgres=False,
+        )
+    )
```

Comment:
Just a nit, up to you to decide on the style for others as well :D
(since we aren't reusing that variable anywhere else, we could just use the string directly)
```suggestion
@pytest.fixture(scope="function", name="flask_app")
def flask_app_fixture(juju: jubilant.Juju, pytestconfig: pytest.Config):
    yield next(
        generate_app_fixture(
            juju=juju,
            pytestconfig=pytestconfig,
            framework=framework,
            image_name="test-flask-image",
            use_postgres=False,
        )
    )
```

Reply:
Its easier to copy paste between other fixture if I keep the `framework` variable :smile: